### PR TITLE
FUSETOOLS-2123 - fixing refactoring issue

### PR DESCRIPTION
**Pull request by bfitzpat**
I don't like the fact that this creates a flashing progress dialog as we rename each package, but it does resolve the issue. That said, there is still a problem with the generated code not building correctly. So I'm open to suggestions here.

### CODE REVIEW

**Comment by bfitzpat** [comment]
Awesome. I'm waiting to see if @apupier or @lhein  have any ideas on the progress dialog. if not, we're good to go. Thanks for taking a look @jpav

**Comment by bfitzpat** [comment]
I'll go ahead and commit this and we can issue course corrections if folks have ideas on how to make it better.

**Comment by lhein** [comment]
Brian, it would have been better to give us a chance to review it.

**Comment by jpav** [comment]
I approved it, and we’re in a bit of a time crunch here.  This fix was holding up two others.

**Comment by lhein** [comment]
I am a bit scared by "the fact that this creates a flashing progress dialog".

**Comment by jpav** [comment]
We’re not crazy about it either, so will revisit later.

File: transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java [hunk 1]
```diff
@@ -497,10 +501,16 @@ private String generateXmlModel(XmlModelGenerator generator,
...
                 JDefinedClass modelClass = classIter.next();
+                IType foundType = project.findType(modelClass.fullName(), monitor);
                 IPackageFragment newPkg = project.findType(modelClass.fullName(), monitor).getPackageFragment();
                 String newPkgName = newPkg.getElementName() + '_' + time;
-                newPkg.rename(newPkgName, true, monitor);
+                
+                // handle refactoring via the refactor framework so we get package renaming
+                // as well as import handling and everything down the line for FUSETOOLS-2123
+                RenameSupport renameSupport = RenameSupport.create(newPkg, newPkgName, RenameSupport.UPDATE_REFERENCES);
+                renameSupport.perform(getShell(), new ProgressMonitorDialog(getShell()));
```
Comment (jpav) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java]: Well, I stand corrected. @bfitzpat, can you take a look at this?

Comment (bfitzpat) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java]: Yup.

**Comment by jpav** [comment]
Not sure if we still have time for this, but we really should create a test for this, too.

File: transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java (hunk 1 above)
Comment (bfitzpat) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java]: Already setting setNeedsProgressMonitor.

https://github.com/fusesource/fuseide/blob/master/transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/extensions/DataMapperEndpointFigureFeature.java#L41

Comment (bfitzpat) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java]: And setting it to the container is MUCH MUCH better. Thanks @apupier !!

**Comment by bfitzpat** [comment]
Unfortunately I don't have time to write a test today. I can definitely do that tomorrow however. I did get Aurelien's fix in - https://github.com/fusesource/fuseide/pull/739

File: transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java [hunk 2]
```diff
@@ -497,10 +501,16 @@ private String generateXmlModel(XmlModelGenerator generator,
             Iterator<JDefinedClass> classIter = modelPkg.classes();
             if (classIter.hasNext()) {
                 JDefinedClass modelClass = classIter.next();
+                IType foundType = project.findType(modelClass.fullName(), monitor);
                 IPackageFragment newPkg = project.findType(modelClass.fullName(), monitor).getPackageFragment();
                 String newPkgName = newPkg.getElementName() + '_' + time;
-                newPkg.rename(newPkgName, true, monitor);
+                
+                // handle refactoring via the refactor framework so we get package renaming
```
Comment (apupier) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java]: I think that this comment should be part of the commit message, not the code.
When reading code we see that it is using the Refactoring Suppport.

File: transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java [hunk 3]
```diff
@@ -497,10 +501,16 @@ private String generateXmlModel(XmlModelGenerator generator,
             Iterator<JDefinedClass> classIter = modelPkg.classes();
             if (classIter.hasNext()) {
                 JDefinedClass modelClass = classIter.next();
+                IType foundType = project.findType(modelClass.fullName(), monitor);
```
Comment (apupier) [inline transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java]: the foundType class is not used

**Pull request merged**